### PR TITLE
[GSoC] Assumptions of Dependence of Rvs

### PR DESCRIPTION
**Pull request by Smit-create**
#### References to other Issues or PRs

Continues #17387

#### Other comments
I will be pushing more commits on this branch by following the reviews on the original PR as well as the discussions in #17260

#### Release Notes

* stats
   * Allowing assumptions between random variables using `Covariance`

ping @Upabjojr @czgdp1807

### CODE REVIEW

File: sympy/stats/rv.py [hunk 1]
```diff
@@ -514,6 +515,33 @@ def conditional_space(self, condition, normalize=True, **kwargs):
 
         return space(domain, density)
 
+class DependentPSpace(PSpace):
+
+    @staticmethod
+    def _create_MutlivariateNormalDistribution(syms, assumps):
```
Comment (eric-wieser) [inline sympy/stats/rv.py:521]:
```suggestion
    def _create_MultivariateNormalDistribution(syms, assumps):
```

File: sympy/stats/tests/test_rv.py [hunk 2]
```diff
@@ -269,6 +269,15 @@ def test_dependent_finite():
     XX, YY = given(Tuple(X, Y), X + Y > 5)  # Create a dependency
     assert dependent(XX, YY)
 
+def test_dependent_pspace():
+    X, Y = Normal('X', 0, 1), Normal('Y', 0, 1)
+    assumps = Eq(Covariance(X, Y), S(1)/2)
+    rv = Dummy('rv', real=True)
+    assert density(X + Y, assumps).dummy_eq(
```
Comment (czgdp1807) [inline sympy/stats/tests/test_rv.py:276]: `assumps` goes as `given`, right?

Comment (Smit-create) [reply sympy/stats/tests/test_rv.py:276]: I didn't get your question. But, currently, `assumps` go as `condition` to `density` and then without going to `given` it passes the call to `DependentPSpace`. Did you mean that?

Comment (czgdp1807) [reply sympy/stats/tests/test_rv.py:276]: Well, as of now, I cannot do, 
```python
In [8]: X = Binomial('X', 5, S(1)/2)                                            

In [9]: Y = Binomial('Y', 6, S(1)/4)                                            

In [12]: P(X > 2, Eq(P(X > 1, Y > 1), S(1)/2))
ValueError: Overlapping Random Variables
```
In other words, queries like P(E|J) where E and J are some events dependent on each other isn't possible?

Comment (Smit-create) [reply sympy/stats/tests/test_rv.py:276]: Actually this will need some fixes after many changes, I will try to update this by tonight.

Comment (czgdp1807) [reply sympy/stats/tests/test_rv.py:276]: Supporting these types of queries can be challenging.

File: sympy/stats/rv.py [hunk 3]
```diff
@@ -514,6 +515,33 @@ def conditional_space(self, condition, normalize=True, **kwargs):
...
+        xrep_dict = {eq.lhs: eq.rhs for eq in assumps.args}
+        mu = [sym.pspace.distribution.args[0] for sym in syms]
+        sigma = ImmutableMatrix(
+                [[Covariance(s1, s2).expand() for s2 in syms] for s1 in syms]
+                ).xreplace(xrep_dict)
+        return Normal('_', mu, sigma).pspace.distribution
+
+
+    @staticmethod
+    def compute_density(expr, assumps):
```
Comment (czgdp1807) [inline sympy/stats/rv.py:535]: What kind of assumptions of dependence does this support, like Covariance or any other metric? What's popular in the literature?

Comment (Smit-create) [reply sympy/stats/rv.py:535]: Currently, I have added the Covariance

**Comment by oscarbenjamin** [comment]
What's the status of this?

**Comment by Smit-create** [comment]
I'm closing this for now and suggest continuing the original work from #17387

**Pull request closed without merging**